### PR TITLE
Add CLI framework files

### PR DESCRIPTION
**Pull request by elliotgunton**
**Pull Request Checklist**
- [x] Work towards #670 
- [ ] Tests added
- [ ] Documentation/examples added
- [ ] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

**Description of PR**
Currently, ...

This PR adds/changes/fixes...

### CODE REVIEW

File: src/hera/_cli/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+"""Hera's command line interface."""
```
Comment (flaviuvadan) [inline src/hera/_cli/__init__.py:1]: I suggest we leave this public as there might be uses for it. Thoughts?

Comment (elliotgunton) [reply src/hera/_cli/__init__.py:1]: Hmm, I think it depends on the API of the CLI, but it should be passing through to public functions we'll add elsewhere. We can always go public if want too, which is my preference unless the "might" becomes definite

**Review by flaviuvadan — COMMENTED** [review]
Looks _awesome_ so far 🎖️ posted a question in our private chat about the organization of the CLI modules

**Review by flaviuvadan — APPROVED** [review]
This LGTM + we aligned on Slack re: the structure + direction of the CLI. Would you like to merge this PR @elliotgunton? Happy to review/facilitate integration in whatever order you'd like to use for delivering this

**Comment by elliotgunton** [comment]
@flaviuvadan I want to get the docs together too (e.g. installation) for a minimal PR, but also leaving this branch to develop this on a feature branch could help? Either way would rather not merge for the moment

**Comment by samj1912** [comment]
I'm going to close this PR for now. We can reopen this once we are working on the CLI.

**Pull request closed without merging**